src/main/java/com/uu_demo/service/impl/model/UserServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.uu_demo.service.impl.model;
+
+import com.uu_demo.dao.abstracts.UserDao;
+import com.uu_demo.models.entity.Role;
+import com.uu_demo.models.entity.User;
+import com.uu_demo.service.GenericServiceAbstract;
+import com.uu_demo.service.abstracts.model.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+
+@Service
+public class UserServiceImpl extends GenericServiceAbstract<User, Long> implements UserService {
```

Comment:
The service layer should contain business logic - not just be a proxy for the data access layer.